initial commit

msteimel47591 opened the pull request.

wkktoria reviewed (changes requested):
Hi @msteimel47591! Good work on completing this project! ⭐ 

I like the approach with creating additional DTOs for flashcards and stack with "clean IDs"!

While testing your application, I saw that it doesn't meet one of the requirements. The requirements say that stacks should have unique names, but I can create stacks with names `Test` and `test`. I think it should be case insensitive. During study sessions answers could also be case insensitive and it would be good to trim any whitespaces.

About your code style, you should you [file scoped namespaces](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces). It allows to save one indentation level. 😄 

I have few more suggestions about your solution:
- Consider initializing reference types (like `string`), for strings you can use [string.Empty](https://learn.microsoft.com/en-us/dotnet/api/system.string.empty). It helps to avoid null reference exceptions.
- It's good to change access modifier to `private` for methods that you don't use in other classes.
- In `switch` statements you don't need the `break` after `return`.
- There are [DateTime](https://learn.microsoft.com/en-us/dotnet/api/system.datetime) and [DateOnly](https://learn.microsoft.com/en-us/dotnet/api/system.dateonly). Don't use strings for dates.
- There is some code duplication in database related classes. You have `dataBaseName`, `dbConnectionString` strings which are the same everywhere. It could be good to extract them.

Feel free to reach me out if you have any questions or need support! ☺️

msteimel47591 commented:
Thanks for the review. I'm still figuring out github, so if this doesn't come out right I apologize. Regarding the requested changes:

1. Added .Trim().ToLower() for case insensitivity of stack names and study session answers.
2. Changed namespaces from namespace Name{} to namespace Name; to remove a level of indentation.
3. I only found one instance, but I changed a string instantiation from using null to using String.Empty.
4. Removed unnecessary break; in the switch statements.
5.  Changed the date field of the study database to use the DATETIME2 datatype instead of NVARCHAR and updated the rest of the application as necessary to accommodate.
6. Removed the duplicated connection strings.

Let me know if I didn't make any of the corrections the way you were expecting. Thanks again.

wkktoria commented on FlashCards_msteimel/FlashCards/Database/DatabaseCreation.cs:
```diff
@@ -1,50 +1,50 @@
 {
-    internal static class DatabaseCreation
+    private static readonly string serverConnectionString = "Server=LAPTOP-LFCOM607;Integrated Security=true;TrustServerCertificate=True;";
+    public static readonly string dataBaseName = "FlashCardsDB";
```
`dataBaseName` can be made `private`, because it is only used in this class.

wkktoria commented on FlashCards_msteimel/FlashCards/Models/Stack.cs:
```diff
@@ -1,8 +1,7 @@
-        public string Name { get; set; }
-    }
+    public int Id { get; set; }
+    public string Name { get; set; }
```
`Name` could be initialized with `string.Empty`.

wkktoria commented on FlashCards_msteimel/FlashCards/Models/StudySession.cs:
```diff
@@ -1,10 +1,9 @@
-        public int StackId { get; set; }
-    }
+    public DateTime Date { get; set; }
+    public string Score { get; set; }
+    public string StackName { get; set; }
```
These could also be initialized with `string.Empty`.

wkktoria commented on FlashCards_msteimel/FlashCards/Models/StudySession.cs:
```diff
@@ -1,10 +1,9 @@
-        public string StackName { get; set; }
-        public int StackId { get; set; }
-    }
+    public DateTime Date { get; set; }
```
Good change to use `DateTime` for date.

wkktoria commented on FlashCards_msteimel/FlashCards/Models/FlashCard.cs:
```diff
@@ -1,10 +1,9 @@
-        public int StackId { get; set; }
-    }
+    public int Id { get; set; }
+    public string Question { get; set; }
+    public string Answer { get; set; }
```
These could be initialized with `string.Empty`.

wkktoria commented on FlashCards_msteimel/FlashCards/Models/FlashCardDto.cs:
```diff
@@ -1,8 +1,7 @@
-        public string Question { get; set; }
-        public string Answer { get; set; }
-    }
+    public string Question { get; set; }
+    public string Answer { get; set; }
```
These could be initialized with `string.Empty`.
